src/main/java/com/hillel/items_exchange/controller/UserController.java

```diff
@@ -0,0 +1,48 @@
+package com.hillel.items_exchange.controller;
+
+import com.hillel.items_exchange.dto.UserDto;
+import com.hillel.items_exchange.model.BaseEntity;
+import com.hillel.items_exchange.service.UserService;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.context.MessageSource;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.access.AccessDeniedException;
+import org.springframework.web.bind.annotation.*;
+
+import java.security.Principal;
+import java.util.Locale;
+
+@RestController
+@RequestMapping("/user")
+@CrossOrigin(origins = "http://localhost:4200")
+@RequiredArgsConstructor
+@Slf4j
+public class UserController {
+
+    private final MessageSource messageSource;
+    private final UserService userService;
+
+    @GetMapping("/info/{id}")
+    public @ResponseBody
+    ResponseEntity<UserDto> getUserInfo(@PathVariable("id") long id, Principal principal) {
```

Comment:
In my opinion `UserDto` doesn't cover all user's data

Reply:
I agree with you. But that's the task. If the task is changed to return all information to the user, I will think about how to return absolutely all the information.

Reply:
@21November would you mind adding validation annotation to id variable?